src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -661,7 +672,10 @@ def _save_last_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Dict[
 
         # set the last model path before saving because it will be part of the state.
         previous, self.last_model_path = self.last_model_path, filepath
-        self._save_checkpoint(trainer, filepath)
+        if self._fs.protocol == "file" and self._last_checkpoint_saved and self.save_top_k != 0:
```

Comment:
Why the ` self.save_top_k != 0` condition? Couldn't "last" checkpoints still track "monitor=None" checkpoints?

Reply:
Hmm, it could be that I'm missing something here due to the complexities, but my understanding is that if we set `save_top_k=0` then save_last=True would mean we save only this file, and so there is nothing to symlink to and the last.ckpt file is just the checkpoint itself